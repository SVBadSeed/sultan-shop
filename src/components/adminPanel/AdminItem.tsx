import React from 'react'

import {useDispatch, useSelector} from "react-redux"
import {deleteCard, getItemId} from "../../redux/filter/FilterSlice"
import {selectCardById} from "../../redux/filter/selectors"
import {removeItem} from "../../redux/cart/CartSlice"

type AdminItemProps = {
    imageUrl: string,
    name: string,
    type: string,
    size: number,
    barcode: number,
    producer: string,
    brand: string,
    description: string,
    price: number,
    typeCare: string[],
    id: string,
    setVisibleEdit: any
}

const AdminItem: React.FC<AdminItemProps> = (props) => {
    const dispatch = useDispatch()
    const itemById = useSelector(selectCardById(props.id))


    const deleteCardHandler = () => {

        if (window.confirm('Вы точно хотите удалить карточку товара?')) {
            dispatch(deleteCard(props.id))
            dispatch(removeItem(props.id))
        }
    }
    const editCardHandler = () => {
        props.setVisibleEdit(true)

        dispatch(getItemId(itemById.id))
    }

    return (
        <div className='admin-panel__item'>
            <div className='admin-panel__content'>
                <div className='admin-panel__content-image'>
                    <span>Картинка</span>
                    <img src={props.imageUrl} alt=""/>
                </div>
                <div className='admin-panel__content-name'>
                    <span>Название</span>
                    <p> {props.name} </p>
                </div>
                <div className='admin-panel__content-text'>
                    <span>Описание</span>
                    <p> {props.description} </p>
                </div>
                <div className='admin-panel__content-size'>
                    <span>Размер</span>
                    <p> {props.size} </p>
                </div>
                <div className='admin-panel__content-type'>
                    <span>Тип</span>
                    <p> {props.type} </p>
                </div>
                <div className='admin-panel__content-typeCare'>
                    <span>Тип ухода</span>
                    <p> {props.typeCare} </p>
                </div>
                <div className='admin-panel__content-brand'>
                    <span>Бренд</span>
                    <p> {props.brand} </p>
                </div>
                <div className='admin-panel__content-producer'>
                    <span>Производитель</span>
                    <p> {props.producer} </p>
                </div>
                <div className='admin-panel__content-price'>
                    <span>Цена</span>
                    <p>{props.price}</p>
                </div>
                <div className='admin-panel__content-barcode'>
                    <span>Штрихкод</span>
                    <p>{props.barcode}</p>
                </div>
            </div>
            <button onClick={deleteCardHandler} className='admin-panel__button-delete'>Удалить карточку</button>
            <button onClick={editCardHandler} className='admin-panel__button-edit'>Редактировать
                карточку
            </button>
        </div>
    )
}

export default AdminItem