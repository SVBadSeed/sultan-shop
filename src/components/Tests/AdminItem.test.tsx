import {render} from "@testing-library/react"
import * as reduxHooks from 'react-redux'
import AdminItem from "../adminPanel/AdminItem"

jest.mock('react-redux')

const mockedUseSelector = jest.spyOn(reduxHooks, 'useSelector')

describe('AdminItem', () => {
    it('AdminItem snapshot', () => {
        mockedUseSelector.mockReturnValue(1)

        const component = render(<AdminItem
            imageUrl='https://sun9-19.userapi.com/impg/2PPRq_DqAos6wV6irSDxgUNzSCz-QaMLN2F30w/d9AlL4fl_eU.jpg?size=213x320&quality=95&sign=917a56004a0253a74d9f63bce0b055be&type=album'
            name='ADIDAS'
            type='мл'
            size={250}
            barcode={4604049097549}
            producer='Adidas'
            brand='ADIDAS'
            description='Купить Гель для душа 250 мл Adidas Уход за телом'
            price={48}
            typeCare={["УХОД ЗА ТЕЛОМ"]}
            id='0'
            setVisibleEdit={() => {
            }}/>)

        expect(component).toMatchSnapshot()
    })
})