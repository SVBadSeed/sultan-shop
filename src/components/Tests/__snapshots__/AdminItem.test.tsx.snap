// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminItem AdminItem snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="admin-panel__item"
      >
        <div
          class="admin-panel__content"
        >
          <div
            class="admin-panel__content-image"
          >
            <span>
              Картинка
            </span>
            <img
              alt=""
              src="https://sun9-19.userapi.com/impg/2PPRq_DqAos6wV6irSDxgUNzSCz-QaMLN2F30w/d9AlL4fl_eU.jpg?size=213x320&quality=95&sign=917a56004a0253a74d9f63bce0b055be&type=album"
            />
          </div>
          <div
            class="admin-panel__content-name"
          >
            <span>
              Название
            </span>
            <p>
               
              ADIDAS
               
            </p>
          </div>
          <div
            class="admin-panel__content-text"
          >
            <span>
              Описание
            </span>
            <p>
               
              Купить Гель для душа 250 мл Adidas Уход за телом
               
            </p>
          </div>
          <div
            class="admin-panel__content-size"
          >
            <span>
              Размер
            </span>
            <p>
               
              250
               
            </p>
          </div>
          <div
            class="admin-panel__content-type"
          >
            <span>
              Тип
            </span>
            <p>
               
              мл
               
            </p>
          </div>
          <div
            class="admin-panel__content-typeCare"
          >
            <span>
              Тип ухода
            </span>
            <p>
               
              УХОД ЗА ТЕЛОМ
               
            </p>
          </div>
          <div
            class="admin-panel__content-brand"
          >
            <span>
              Бренд
            </span>
            <p>
               
              ADIDAS
               
            </p>
          </div>
          <div
            class="admin-panel__content-producer"
          >
            <span>
              Производитель
            </span>
            <p>
               
              Adidas
               
            </p>
          </div>
          <div
            class="admin-panel__content-price"
          >
            <span>
              Цена
            </span>
            <p>
              48
            </p>
          </div>
          <div
            class="admin-panel__content-barcode"
          >
            <span>
              Штрихкод
            </span>
            <p>
              4604049097549
            </p>
          </div>
        </div>
        <button
          class="admin-panel__button-delete"
        >
          Удалить карточку
        </button>
        <button
          class="admin-panel__button-edit"
        >
          Редактировать карточку
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="admin-panel__item"
    >
      <div
        class="admin-panel__content"
      >
        <div
          class="admin-panel__content-image"
        >
          <span>
            Картинка
          </span>
          <img
            alt=""
            src="https://sun9-19.userapi.com/impg/2PPRq_DqAos6wV6irSDxgUNzSCz-QaMLN2F30w/d9AlL4fl_eU.jpg?size=213x320&quality=95&sign=917a56004a0253a74d9f63bce0b055be&type=album"
          />
        </div>
        <div
          class="admin-panel__content-name"
        >
          <span>
            Название
          </span>
          <p>
             
            ADIDAS
             
          </p>
        </div>
        <div
          class="admin-panel__content-text"
        >
          <span>
            Описание
          </span>
          <p>
             
            Купить Гель для душа 250 мл Adidas Уход за телом
             
          </p>
        </div>
        <div
          class="admin-panel__content-size"
        >
          <span>
            Размер
          </span>
          <p>
             
            250
             
          </p>
        </div>
        <div
          class="admin-panel__content-type"
        >
          <span>
            Тип
          </span>
          <p>
             
            мл
             
          </p>
        </div>
        <div
          class="admin-panel__content-typeCare"
        >
          <span>
            Тип ухода
          </span>
          <p>
             
            УХОД ЗА ТЕЛОМ
             
          </p>
        </div>
        <div
          class="admin-panel__content-brand"
        >
          <span>
            Бренд
          </span>
          <p>
             
            ADIDAS
             
          </p>
        </div>
        <div
          class="admin-panel__content-producer"
        >
          <span>
            Производитель
          </span>
          <p>
             
            Adidas
             
          </p>
        </div>
        <div
          class="admin-panel__content-price"
        >
          <span>
            Цена
          </span>
          <p>
            48
          </p>
        </div>
        <div
          class="admin-panel__content-barcode"
        >
          <span>
            Штрихкод
          </span>
          <p>
            4604049097549
          </p>
        </div>
      </div>
      <button
        class="admin-panel__button-delete"
      >
        Удалить карточку
      </button>
      <button
        class="admin-panel__button-edit"
      >
        Редактировать карточку
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CatalogParameters CatalogParameters snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="admin-panel__item"
      >
        <div
          class="admin-panel__content"
        >
          <div
            class="admin-panel__content-image"
          >
            <span>
              Картинка
            </span>
            <img
              alt=""
              src="https://sun9-19.userapi.com/impg/2PPRq_DqAos6wV6irSDxgUNzSCz-QaMLN2F30w/d9AlL4fl_eU.jpg?size=213x320&quality=95&sign=917a56004a0253a74d9f63bce0b055be&type=album"
            />
          </div>
          <div
            class="admin-panel__content-name"
          >
            <span>
              Название
            </span>
            <p>
               
              ADIDAS
               
            </p>
          </div>
          <div
            class="admin-panel__content-text"
          >
            <span>
              Описание
            </span>
            <p>
               
              Купить Гель для душа 250 мл Adidas Уход за телом
               
            </p>
          </div>
          <div
            class="admin-panel__content-size"
          >
            <span>
              Размер
            </span>
            <p>
               
              250
               
            </p>
          </div>
          <div
            class="admin-panel__content-type"
          >
            <span>
              Тип
            </span>
            <p>
               
              мл
               
            </p>
          </div>
          <div
            class="admin-panel__content-typeCare"
          >
            <span>
              Тип ухода
            </span>
            <p>
               
              УХОД ЗА ТЕЛОМ
               
            </p>
          </div>
          <div
            class="admin-panel__content-brand"
          >
            <span>
              Бренд
            </span>
            <p>
               
              ADIDAS
               
            </p>
          </div>
          <div
            class="admin-panel__content-producer"
          >
            <span>
              Производитель
            </span>
            <p>
               
              Adidas
               
            </p>
          </div>
          <div
            class="admin-panel__content-price"
          >
            <span>
              Цена
            </span>
            <p>
              48
            </p>
          </div>
          <div
            class="admin-panel__content-barcode"
          >
            <span>
              Штрихкод
            </span>
            <p>
              4604049097549
            </p>
          </div>
        </div>
        <button
          class="admin-panel__button-delete"
        >
          Удалить карточку
        </button>
        <button
          class="admin-panel__button-edit"
        >
          Редактировать карточку
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="admin-panel__item"
    >
      <div
        class="admin-panel__content"
      >
        <div
          class="admin-panel__content-image"
        >
          <span>
            Картинка
          </span>
          <img
            alt=""
            src="https://sun9-19.userapi.com/impg/2PPRq_DqAos6wV6irSDxgUNzSCz-QaMLN2F30w/d9AlL4fl_eU.jpg?size=213x320&quality=95&sign=917a56004a0253a74d9f63bce0b055be&type=album"
          />
        </div>
        <div
          class="admin-panel__content-name"
        >
          <span>
            Название
          </span>
          <p>
             
            ADIDAS
             
          </p>
        </div>
        <div
          class="admin-panel__content-text"
        >
          <span>
            Описание
          </span>
          <p>
             
            Купить Гель для душа 250 мл Adidas Уход за телом
             
          </p>
        </div>
        <div
          class="admin-panel__content-size"
        >
          <span>
            Размер
          </span>
          <p>
             
            250
             
          </p>
        </div>
        <div
          class="admin-panel__content-type"
        >
          <span>
            Тип
          </span>
          <p>
             
            мл
             
          </p>
        </div>
        <div
          class="admin-panel__content-typeCare"
        >
          <span>
            Тип ухода
          </span>
          <p>
             
            УХОД ЗА ТЕЛОМ
             
          </p>
        </div>
        <div
          class="admin-panel__content-brand"
        >
          <span>
            Бренд
          </span>
          <p>
             
            ADIDAS
             
          </p>
        </div>
        <div
          class="admin-panel__content-producer"
        >
          <span>
            Производитель
          </span>
          <p>
             
            Adidas
             
          </p>
        </div>
        <div
          class="admin-panel__content-price"
        >
          <span>
            Цена
          </span>
          <p>
            48
          </p>
        </div>
        <div
          class="admin-panel__content-barcode"
        >
          <span>
            Штрихкод
          </span>
          <p>
            4604049097549
          </p>
        </div>
      </div>
      <button
        class="admin-panel__button-delete"
      >
        Удалить карточку
      </button>
      <button
        class="admin-panel__button-edit"
      >
        Редактировать карточку
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
